lib/Conversion/TritonGPUToLLVM/HistogramOpToLLVM.cpp

```diff
@@ -25,7 +25,12 @@ static SmallVector<Value> computeWarpLevelHistogram(
   int numBits = llvm::Log2_64(numBins);
   int numBitsLaneId = llvm::Log2_64(numThreadPerWarp);
   unsigned numElementsPerThreads = getTotalElemsPerThread(srcType);
-  unsigned numThreadWithUniqueData = getThreadsPerWarp(srcType)[0];
+  auto numThreadWithUniqueData = getThreadsPerWarp(srcType);
+  auto replicationFactor =
+      numThreadPerWarp / std::accumulate(numThreadWithUniqueData.begin(),
+                                         numThreadWithUniqueData.end(), 1,
+                                         std::multiplies<>());
```

Comment:
technically there could be redundant data across warps. That wouldn't handle it?

Reply:
I think you could apply the same trick to the final result?